sharktank/sharktank/layers/paged_llama_attention_block.py

```diff
@@ -214,6 +229,30 @@ def forward(
             x, start_index, embedding, embedding_batch_mask
         )
 
+        if self.use_qk_norm:
+            xq = self.qk_norm(xq)
+            xk = self.qk_norm(xk)
+
+        # Use temperature tuning from https://arxiv.org/abs/2501.19399
+        # Ken M. Nakanishi - Scalable-Softmax Is Superior for Attention (2025)
+        if self.attn_temperature_tuning and not self.use_rope:
+            if start_positions is None:
+                cache_position = torch.arange(0, h.shape[1], dtype=torch.long)
```

Comment:
This is a part that would need to be refactored in a subsequent PR. There are questions like, are we going to use padding at the start and then the decode step would need to be supported.
Nevertheless, `cache_position` is not the right name for this. I think it will get confused with the KV cache. We are here essentially taking the batch sequence length.

Reply:
Long-term it would probably have to be an arange over the current accumulated batch sequence length across all prefill/decode invocations.

Reply:
But the name was adopted from Transformers only.